doc/GLCM.ipynb

```diff
@@ -0,0 +1,414 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# get Imglib2 resources\n",
+    "MatrixOrientation2D = imagej.sj.jimport('net.imagej.ops.image.cooccurrenceMatrix.MatrixOrientation2D')\n",
+    "\n",
+    "# setup orientation dictionary\n",
+    "orientations = {\n",
+    "    \"antidiagonal\": MatrixOrientation2D.ANTIDIAGONAL,\n",
+    "    \"diagonal\"    : MatrixOrientation2D.DIAGONAL,\n",
+    "    \"horizontal\"  : MatrixOrientation2D.HORIZONTAL,\n",
+    "    \"vertical\"    : MatrixOrientation2D.VERTICAL\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The following cell will setup the `run_glcm()` and `process_crops()`functions. The `run_glcm()` function accepts a single 2D image, obtains the _correlation_ and _difference variance_ GLCM texture values and returns a tuple. The `process_crops()` function will process a list of crops we will generate later and feed each crop to the `run_glcm()` function and store it's output in a `pandas.DataFrame`. Once we have the `pandas.DataFrame` we can easily plot the data on a scattterplot."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def run_glcm(img, gray_levels: int, dist: int, angle: str):\n",
+    "    \"\"\"\n",
+    "    Compute the correlation and difference variance GLCM textures from an image.\n",
+    "\n",
+    "    :param img: An input ImgPlus\n",
+    "    :param gray_levels: Number of gray levels\n",
+    "    :param dist: Distance in pixels\n",
+    "    :param angle: Angle to analyze image\n",
+    "    :return: A tuple of floats: (correlation, difference variance)\n",
+    "    \"\"\"\n",
+    "    # compute correlation and difference variance\n",
+    "    corr = ij.op().haralick().correlation(img, gray_levels, dist, orientations.get(angle))\n",
+    "    diff = ij.op().haralick().differenceVariance(img, gray_levels, dist, orientations.get(angle))\n",
+    "    \n",
+    "    # convert to Python float\n",
+    "    corr = ij.py.from_java(corr)\n",
+    "    diff = ij.py.from_java(diff)\n",
+    "\n",
+    "    return (corr.value, diff.value)\n",
+    "\n",
+    "def process_crops(crops, gray_levels: int, dist: int, angle: str) -> pd.DataFrame:\n",
+    "    \"\"\"\n",
+    "    Process a list of ImgPlus images with GLCM texture analysis.\n",
+    "\n",
+    "    :param crops: A list of ImgPlus images\n",
+    "    :param gray_levels: Number of gray levels\n",
+    "    :param dist: Distance in pixels\n",
+    "    :param angle: Angle to analyze image\n",
+    "    :return: Pandas DataFrame\n",
+    "    \"\"\"\n",
+    "    # compute the correlation and difference variance for each crop\n",
+    "    glcm_results = []\n",
+    "    for c in crops:\n",
```

Comment:
Yeah I think you saw what I was originally intending on doing...which was to iterate through all the orientations and average the `correlation` and `differenceVariance` values. This is typically done for textures that don't have any given angle to it. In this case I just used `MatrixOrientation2D.HORIZONTAL` because it was easy.

Another point that I think is important, even though that I'm not using all the angles that `MatrixOrientation2D` offers I think its important to show users what **is available**. From that perspective I'd argue to keep the `orientations` regardless.